src/order/complete_lattice.lean

```diff
@@ -473,6 +473,19 @@ le_antisymm
   (Sup_le $ assume b h, le_supr_of_le b $ le_supr _ h)
   (supr_le $ assume b, supr_le $ assume h, le_Sup h)
 
+-- Unfortunately `ι : Sort u_4` doesn't work with `set ι`
+theorem bsupr_eq_supr {ι : Type*} {s : ι → α}
+  {p : set ι} : (⨆ i (H : p i), s i) = ⨆ (i : p), s i :=
```

Comment:
```suggestion
  {p : set ι} : (⨆ i (H : i ∈ p), s i) = ⨆ (i : p), s i :=
```
or
```suggestion
  {p : ι → Prop} : (⨆ i (H : p i), s i) = ⨆ (i : subtype p), s i :=
```
Don't ever use `s x` to mean `s ∈ x`, it's not the canonical spelling and leaves the caller a mess to clean up

Reply:
We might already have this lemma in one of those spellings

Reply:
Yes, this is just `supr_subtype`